src/parser/jobs/brd/modules/RagingStrikes.tsx

```diff
@@ -0,0 +1,131 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import {RotationTargetOutcome} from 'components/ui/RotationTable'
+import {Action} from 'data/ACTIONS'
+import {ActionRoot} from 'data/ACTIONS/root'
+import STATUSES from 'data/STATUSES'
+import {BuffEvent, CastEvent} from 'fflogs'
+import {BuffWindowModule, BuffWindowState, BuffWindowTrackedAction} from 'parser/core/modules/BuffWindow'
+import {SEVERITY} from 'parser/core/modules/Suggestions'
+import React from 'react'
+import {isDefined} from 'utilities'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+const SUPPORT_ACTIONS: Array<keyof ActionRoot> = [
+	'ARMS_LENGTH',
+	'FOOT_GRAZE',
+	'HEAD_GRAZE',
+	'LEG_GRAZE',
+	'NATURES_MINNE',
+	'PELOTON',
+	'REPELLING_SHOT',
+	'SECOND_WIND',
+	'SPRINT',
+	'THE_WARDENS_PAEAN',
+	'TROUBADOUR',
+]
+
+export default class RagingStrikes extends BuffWindowModule {
+	static handle = 'rs'
+	static title = t('brd.rs.title')`Raging Strikes`
+	static displayOrder = DISPLAY_ORDER.RAGING_STRIKES
+
+	buffAction = this.data.actions.RAGING_STRIKES
+	buffStatus = STATUSES.RAGING_STRIKES
+
+	private museTimestamps: number[] = []
+	private SUPPORT_ACTIONS: number[] = []
+
+	expectedGCDs = {
+		expectedPerWindow: 8,
+		suggestionContent: <Trans id="brd.rs.suggestions.missedgcd.content">
+			Try to land 8 GCDs (9 GCDs with <StatusLink {...STATUSES.ARMYS_MUSE}/>) during every <ActionLink {...this.data.actions.RAGING_STRIKES}/> window.
+		</Trans>,
+		severityTiers: {
+			1: SEVERITY.MINOR,
+			3: SEVERITY.MEDIUM,
+			5: SEVERITY.MAJOR,
+		},
+	}
+
+	trackedActions = {
+		icon: this.data.actions.BARRAGE.icon,
+		actions: [
+			{
+				action: this.data.actions.BARRAGE,
+				expectedPerWindow: 1,
+			},
+			{
+				action: this.data.actions.IRON_JAWS,
+				expectedPerWindow: 1,
+			},
+		],
+		suggestionContent: <Trans id="brd.rs.suggestions.trackedactions.content">
+			One use of <ActionLink {...this.data.actions.BARRAGE}/> and one use of <ActionLink {...this.data.actions.IRON_JAWS}/> should occur during every <ActionLink {...this.data.actions.RAGING_STRIKES}/> window.
+		</Trans>,
+		severityTiers: {
+			1: SEVERITY.MINOR,
+			3: SEVERITY.MEDIUM,
+			5: SEVERITY.MAJOR,
+		},
+	}
+
+	protected init() {
+		super.init()
+
+		this.SUPPORT_ACTIONS = SUPPORT_ACTIONS.map(actionKey => this.data.actions[actionKey].id)
+		this.addEventHook('applybuff', {to: 'player', abilityId: [STATUSES.ARMYS_MUSE.id]}, this.onApplyMuse)
+	}
+
+	private onApplyMuse = (event: BuffEvent) => this.museTimestamps.push(event.timestamp)
+
+	protected considerAction = (action: Action) => !this.SUPPORT_ACTIONS.includes(action.id)
+
+	protected changeExpectedGCDsClassLogic(buffWindow: BuffWindowState): number {
+		// Expect one extra GCD if we had muse up for this RS
+		const museDuration = STATUSES.ARMYS_MUSE.duration * 1000
+		const isMuseUp = this.museTimestamps
```

Comment:
This seems too aggressive.  I'm a little dubious of adding 1 GCD for Muse even if the Muse buff was applied within the first 10 seconds of Raging Strikes, since if it's only a 1-3 stack muse, they're gaining less than 0.1s per GCD and only going to gain for ~4 GCDs depending on where in the weave window they switched songs and whether Muse follows the Blood Weapon school of timing or the more conventional "10.75 seconds" timing.  

But this is also catching a case like the 06:57 window here: http://localhost:3000/fflogs/dFAy6pwP4L9vRbmM/42/2020 -- Muse happened with the song switch at 06:51 and only caught 2 GCDs at the beginning of the Raging Strikes window.

Reply:
At infinitestory's suggestion I changed the approach here to require 3 GCDs under muse in order to expect another GCD under RS. There's going to be a lot of 9/8s as a result, but I guess it's better to err on the side of 9/8 instead of 8/9.